arch/xtensa/src/common/xtensa_dumpstate.c

```diff
@@ -295,12 +295,11 @@ void xtensa_dumpstate(void)
 
   if (CURRENT_REGS)
     {
-      memcpy(rtcb->xcp.regs,
-             (uintptr_t *)CURRENT_REGS, XCPTCONTEXT_SIZE);
+      rtcb->xcp.regs = (uint32_t *)CURRENT_REGS;
     }
   else
     {
-      up_saveusercontext(rtcb->xcp.regs);
+      up_saveusercontext(&rtcb->xcp.regs);
```

Comment:
need keep the prototype in:
include/nuttx/arch.h

Reply:
The prototype is only there.  However it takes a `void *`, so we have some flexibility.

Reply:
fputest code assume it a pointer, not a pointer of pointer.

Reply:
Ah, I forgot about this test.

Reply:
Ok, let's address this issue in your cp state patch.